components/callbacks.py

```diff
@@ -161,62 +172,77 @@ def help_callback(update: Update, context: CallbackContext) -> None:
 
 
 def off_on_topic(update: Update, context: CallbackContext) -> None:
+    # Minimal effort LRU cache
+    parsed_messages = cast(Dict, context.chat_data).setdefault(
```

Comment:
Not quite. The idea is: is the original message already contained the `off/on-topic` tag, then we don't need to redirect again. So I store all messages that lead to redirection, but only up to 64. This assumes that it's unlikely that on old message get's edited. If you edit a message that already says "off-topic" but is no longer in this list, then it will redirect again. I'll elaborate the comment a bit

Reply:
ahh I see. Happens often in groups good change